src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -135,15 +169,19 @@ private static String generateXpathQuery(DetailAST ast) {
      * @param root {@code DetailAST} root ast
      * @param lineNumber line number
      * @param columnNumber column number
+     * @param fileText {@code FileText} object
+     * @param tabWidth distance between tab stop position
      * @return list of nodes matching defined line and column number
      */
     private static List<DetailAST> getMatchingAstElements(DetailAST root, int lineNumber,
-                                                          int columnNumber) {
+                                                          int columnNumber, FileText fileText,
+                                                          int tabWidth) {
         final List<DetailAST> result = new ArrayList<>();
         DetailAST curNode = root;
         while (curNode != null && curNode.getLineNo() <= lineNumber) {
             if (curNode.getLineNo() == lineNumber
```

Comment:
extract all conditions to a method. The condition block became difficult to read and understand

Reply:
done